modules/repository/branch.go

```diff
@@ -13,8 +14,45 @@ import (
 	"code.gitea.io/gitea/modules/git"
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
 )
 
+// UpdateBranch updates the branch information in the database. If the branch exist, it will update latest commit of this branch information
+// If it doest not exist, insert a new record into database
+func UpdateBranch(ctx context.Context, repoID, pusherID int64, branchName string, commit *git.Commit) error {
+	cnt, err := git_model.UpdateBranch(ctx, repoID, pusherID, branchName, commit)
```

Comment:
That function is completely wrong here.

Reply:
Why do you think that?

Reply:
OK. I move the function to service layer and renamed function name to `syncBranchToDB`. Maybe it fixed your problem?